### PR TITLE
Update to use the correct network authority for certain operations

### PR DESCRIPTION
**Pull request by william-brooks**
Currently the seed derivation and signing authority for managing gatekeepers are in conflict. This update allows for a dedication `subject` used for seed derivation and fixes it so that an auth_key on the network is required as a signer for these operations.

PR: https://github.com/identity-com/on-chain-identity-gateway/pull/411

Signed-off-by: William Brooks <william@identity.org>
Reviewed-by: Julian Spring <springjulian@gmail.com>

### CODE REVIEW

File: solana/program_v2/packages/tests/src/pass-suite/util.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ export const createNetworkService = async (
 
   return NetworkService.buildFromAnchor(
     program,
+    authority.publicKey, // TODO: Replace with network authority
```
Comment (william-brooks) [inline solana/program_v2/packages/tests/src/pass-suite/util.ts:48]: This file is clearly not used... will check and potentiall remove it

File: solana/program_v2/packages/tests/src/pass-suite/verify.test.ts [hunk 2]
```diff
@@ -169,9 +169,7 @@ describe('Verify a pass', () => {
   });
 
   it('Fails to verify a pass in a halted gatekeeper', async () => {
-    await networkService
-      .setGatekeeperState(networkAuthority.publicKey, GatekeeperState.Halted)
-      .rpc();
+    await networkService.setGatekeeperState(GatekeeperState.Halted).rpc();
```
Comment (william-brooks) [inline solana/program_v2/packages/tests/src/pass-suite/verify.test.ts:172]: This shouldn't work I think... double checking

Comment (william-brooks) [reply solana/program_v2/packages/tests/src/pass-suite/verify.test.ts:172]: Fixed

File: solana/program_v2/programs/gateway_v2/src/state/gatekeeper.rs [hunk 3]
```diff
@@ -17,7 +17,7 @@ pub struct Gatekeeper {
     /// The version of this struct, should be 0 until a new version is released
     pub version: u8,
     /// the authority for this gatekeeper
-    pub authority: Pubkey,
+    pub subject: Pubkey,
```
[the remark this answers is not in the document]
Comment (william-brooks) [reply solana/program_v2/programs/gateway_v2/src/state/gatekeeper.rs:20]: Yeah this was the signer that created the GK initially... but, the network is meant to be the signer... "subject" may not be the correct term, but keeping in pattern with our pass account language... it's still better than `authority` ?

File: solana/program_v2/packages/tests/src/gatekeeper-suite/close-gatekeeper.test.ts [hunk 4]
```diff
@@ -101,12 +105,8 @@ describe('Gateway v2 Client', () => {
 
       // runs closeGatekeeper
       await networkService
-        .closeGatekeeper(
-          networkAuthority.publicKey,
-          undefined,
-          adminAuthority.publicKey
-        )
-        .withPartialSigners(adminAuthority)
+        .closeGatekeeper()
```
[the remark this answers is not in the document]
Comment (william-brooks) [reply solana/program_v2/packages/tests/src/gatekeeper-suite/close-gatekeeper.test.ts:108]: Yes.. thanks!

**Review by JulianSpring — APPROVED** [review]

**Pull request merged**